File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.cs
using System;

public class Solution 
{
    public int solution(int a, int b) 
    {
        int oddCount = 0;
        if(a%2 == 1)
        {
            oddCount++;
        }
        if(b%2 == 1)
        {
            oddCount++;
        }
        
        int answer = 0;
        if(oddCount == 2)
        {
            answer = a * a + b * b;
        }
        else if(oddCount == 1)
        {
            answer = 2 * (a + b);
        }
        else
        {
            answer = Math.Abs(a-b);
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int a, int b) 
{
    if(a%2 == 1 && b%2 == 1)
    {
        return a*a + b*b;
    }
    else if((a%2 == 1 && b%2 != 1) || (a%2 != 1 && b%2 == 1))
    {
        return 2*(a + b);
    }
    else if(a%2 == 0 && b%2 == 0)
    {
        return abs(a-b);
    }
}